hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -215,9 +215,14 @@ loadCradle iniDynFlags (NewCradle fp) def action = do
             -- it on a save whilst there are errors. Subsequent loads won't
             -- be that slow, even though the cradle isn't cached because the
             -- `.hi` files will be saved.
-          Right () -> do
+          Right BIOS.Succeeded -> do
             setCurrentCradle cradle
-            logm $ "Cradle set succesfully"
+            logm "Cradle set succesfully"
+            IdeResultOk <$> action
+
+          Right BIOS.Failed -> do
```

Comment:
@mpickering Not sure how to handle this correctly, especially in combination with https://github.com/mpickering/haskell-ide-engine/pull/79/files#diff-fb69f899fd005326a13dddab5748049fR199

Reply:
Mmm, i would say that cradle did load succesfully in case of SourceError

Reply:
dunno what reason might exist that BIOS.Failed is returned. I dont know `loadTargets'` enough.

Reply:
Should this just return a `IdeResultFail`?

Reply:
No, it is probably fine as it is. It means, that the we could not load the whole module graph, but we can still cache every module up to the point some module failed to typecheck.